File: test/02_stub_test.rb
# 👩‍🏫 02_stub_test missions:
#    - Make the tests pass by using one or more stub test doubles
#    - Add test cases for guesses that are too low and too high

require "minitest/autorun"
require "02_stub"

# ✍️ WRITE YOUR TEST DOUBLE HERE
class InputStub
  def initialize
    raise "please implement me"
  end
end

# ✍️ END

# 🙅‍♀️ DO NOT CHANGE THIS CODE - START

class GuessingGameStubTest < Minitest::Test
  def test_will_return_found_if_the_guess_is_correct
    # Arrange
    number = 50
    input = InputStub.new
    subject = GuessingGameStub.new(number, input)

    # Act
    result = subject.guess

    # Assert
    assert_equal FOUND, result
  end

  # 🙅‍♀️ DO NOT CHANGE THIS CODE - END

  # ✍️ WRITE YOUR TESTS HERE

  def test_will_return_too_low_if_the_guess_is_too_low
    fail "TODO: implement me"
  end

  def test_will_return_too_high_if_the_guess_is_too_high
    fail "TODO: implement me"
  end
end

# ✍️ END

File: javascript/src/02_stub.test.js
const GuessingGame = require("./guessingGame");

class InputStub {
  constructor() {
    throw new Error("not implemented");
  }
}

describe("Guessing Game, Stub Test Double", () => {
  test("will return FOUND if the guess is correct", () => {
    // Arrange
    const number = 50;
    const input = new InputStub();
    const subject = GuessingGame(input, number);

    // Act
    result = subject.guess();

    // Assert
    expect(result).toEqual("FOUND");
  });

  test("will return TOO_LOW if the guess is too low", () => {
    throw new Error("TODO: implement me");
  });

  test("will return TOO_HIGH if the guess is too high", () => {
    throw new Error("TODO: implement me");
  });
});
